src/issuers-verifier.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  openAttestationVerifiers,
+  verificationBuilder,
+  VerificationFragmentType,
+  VerificationManagerOptions,
+  Verifier,
+  CodedError,
+} from "@govtechsg/oa-verify";
+import { getData, SignedWrappedDocument, utils, v2, v3, WrappedDocument } from "@govtechsg/open-attestation";
+
+export enum VerifyAllowedIssuersCode {
+  UNEXPECTED_ERROR = 0,
+  INVALID_IDENTITY = 1,
+  SKIPPED = 2,
+  UNSUPPORTED_V3_DOCUMENT = 3,
+}
+
+const name = "VerifyAllowedIssuers";
+const type: VerificationFragmentType = "ISSUER_IDENTITY";
+const isWrappedV2Document = (document: any): document is WrappedDocument<v2.OpenAttestationDocument> => {
+  return document.data && document.data.issuers;
+};
+const whitelistedIssuers = ["gov.sg", "openattestation.com"];
+export const verifyAllowedIssuers: Verifier<
+  | WrappedDocument<v2.OpenAttestationDocument>
+  | WrappedDocument<v3.OpenAttestationDocument>
+  | SignedWrappedDocument<v2.OpenAttestationDocument>,
+  VerificationManagerOptions,
+  Array<string | undefined>
+> = {
+  skip: () => {
+    return Promise.resolve({
+      status: "SKIPPED",
+      type,
+      name,
+      reason: {
+        code: VerifyAllowedIssuersCode.SKIPPED,
+        codeString: VerifyAllowedIssuersCode[VerifyAllowedIssuersCode.SKIPPED],
+        message: `Document issuers doesn't have "documentStore" / "tokenRegistry" property or doesn't use ${v3.IdentityProofType.DNSTxt} type`,
```

Comment:
Hmm this message is inappropriate but its hard to think of better ones... maybe "Document is not checked with the whitelisted domains"?

Reply:
LOL.GOV.SG should be allowed
lol.powndergov.sg should fail

Reply:
will throw error, if we always run the verifier